src/IrcConnection/IrcConnection.cs

```diff
@@ -50,44 +45,29 @@ public class IrcConnection
     {
         private string           _VersionNumber;
         private string           _VersionString;
+        private IIrcTransportManager _Transport;
         private string[]         _AddressList = {"localhost"};
         private int              _CurrentAddress;
         private int              _Port;
-        private bool             _UseSsl;
-        private bool             _ValidateServerCertificate;
-        private X509Certificate  _SslClientCertificate;
-        private StreamReader     _Reader;
-        private StreamWriter     _Writer;
-        private ReadThread       _ReadThread;
-        private WriteThread      _WriteThread;
+        private ConcurrentQueue<string> _ReadQueue;
+        private AutoResetEvent   _ReadQueueEvent;
+        private WriteManager  _WriteManager;
         private IdleWorkerThread _IdleWorkerThread;
-        private TcpClient        _TcpClient;
-        private Hashtable        _SendBuffer = Hashtable.Synchronized(new Hashtable());
         private int              _SendDelay = 200;
-        private bool             _IsRegistered;
-        private bool             _IsConnected;
-        private bool             _IsConnectionError;
         private bool             _IsDisconnecting;
+        private bool             _IsRegistered;
         private int              _AutoRetryAttempt;
         private bool             _AutoRetry;
         private int              _AutoRetryDelay = 30;
         private int              _AutoRetryLimit = 3;
         private bool             _AutoReconnect;
-        private Encoding         _Encoding = Encoding.Default;
-        public bool EnableUTF8Recode { get; set; }
-        private int              _SocketReceiveTimeout  = 600;
-        private int              _SocketSendTimeout = 600;
-        private int              _IdleWorkerInterval = 60;
+        private int              _IdleWorkerInterval = 5;
         private int              _PingInterval = 60;
-        private int              _PingTimeout = 300;
+        private int              _PingTimeout = 10;
```

Comment:
Why changing the default values? 10 seconds is pretty normal on mobile networks with packet loss